src/main/java/net/dries007/tfc/util/Helpers.java

```diff
@@ -505,9 +508,38 @@ public static void consumeInStackSizeIncrements(ItemStack stack, int totalCount,
         }
     }
 
+    public static void gatherAndConsumeItems(Level level, AABB bounds, IItemHandler inventory, int minSlotInclusive, int maxSlotInclusive)
+    {
+        gatherAndConsumeItems(level.getEntitiesOfClass(ItemEntity.class, bounds, EntitySelector.ENTITY_STILL_ALIVE), inventory, minSlotInclusive, maxSlotInclusive);
+    }
+
+    public static void gatherAndConsumeItems(Collection<ItemEntity> items, IItemHandler inventory, int minSlotInclusive, int maxSlotInclusive)
+    {
+        final List<ItemEntity> availableItemEntities = new ArrayList<>();
+        int availableItems = 0;
+        int availableSlots = 0;
+        for (int slot = minSlotInclusive; slot <= maxSlotInclusive; slot++)
+        {
+            if (inventory.getStackInSlot(slot).isEmpty())
+            {
+                availableSlots++;
+            }
+        }
```

Comment:
Since we do `insertSlots().isEmpty()`, we shouldn't have to do this "available slots" thing. Just pass in the available items, and `safelyConsumeItemsFromEntitiesIndividually` will stop as soon as it can.